Source/ASDisplayNode.mm

```diff
@@ -2480,6 +2494,11 @@ - (void)insertSubnode:(ASDisplayNode *)subnode belowSubnode:(ASDisplayNode *)bel
 - (void)_insertSubnode:(ASDisplayNode *)subnode belowSubnode:(ASDisplayNode *)below
```

Comment:
I thought that the method called above this, _insertSubnode:atSubnodeIndex:sublayerIndex:andRemoveSubnode: was the true choke point - though maybe we need to create an earlier choke point where we can safely acquire all the locks?

I'll assume this part of the patch isn't complete yet, which is fine to worry about later.